lua/starfall/libs_sh/darkrp.lua

```diff
@@ -0,0 +1,1089 @@
+		if not checksafety(success) then return false end
+		if instance.player ~= SF.Superuser then
+			if not SF.Permissions.hasAccess(instance, nil, "darkrp.lockpickHooks") then return false end
+			if instance.player ~= ply then return false end
+		end
+		return true, {
+			ply and instance.Types.Player.Wrap(ply) or nil,
+			success,
+			ent and instance.Types.Entity.Wrap(ent) or nil
+		}
+	end)
+end
+
+--- Functions relating to DarkRP. These functions WILL NOT EXIST if DarkRP is not in use.
+-- @name darkrp
+-- @class library
+-- @libtbl darkrp_library
+SF.RegisterLibrary("darkrp")
+
+return function(instance)
+
+if not DarkRP then return end
+
+local darkrp_library = instance.Libraries.darkrp
+local ply_meta = instance.Types.Player
+local player_methods, plywrap, plyunwrap, getply = ply_meta.Methods, ply_meta.Wrap, ply_meta.Unwrap
+local ent_meta = instance.Types.Entity
+local ents_methods, ewrap, eunwrap, getent = ent_meta.Methods, ent_meta.Wrap, ent_meta.Unwrap
+instance:AddHook("initialize", function()
+	getent = instance.Types.Entity.GetEntity
+	getply = instance.Types.Player.GetPlayer
+end)
+local checkpermission = instance.player == SF.Superuser and function() end or SF.Permissions.check
+
+--- Format a number as a money value. Includes currency symbol.
+-- @param number amount The money to format, e.g. 100000.
+-- @return string The money as a nice string, e.g. "$100,000".
+function darkrp_library.formatMoney(n)
+	checkluatype(n, TYPE_NUMBER)
+	return assertsafety(DarkRP.formatMoney(n))
+end
+
+--- Get the available vehicles that DarkRP supports.
+-- @return table Names, models and classnames of all supported vehicles.
+function darkrp_library.getAvailableVehicles()
+	return instance.Sanitize(DarkRP.getAvailableVehicles())
+end
+
+--- Get all categories for all F4 menu tabs, including all jobs and every entity available for purchase.
+-- @return table All categories.
+function darkrp_library.getCategories()
+	return instance.Sanitize(DarkRP.getCategories())
+end
+
+--- Get all food items.
+-- @return table? Table with food items, or nil if there are none.
+function darkrp_library.getFoodItems()
+	local tbl = DarkRP.getFoodItems()
+	return tbl and instance.Sanitize(tbl) or nil
+end
+
+--- Get the table of all current laws.
+-- @return table A table of all current laws.
+function darkrp_library.getLaws()
+	return instance.Sanitize(DarkRP.getLaws())
+end
+
+--- Get a list of possible shipments.
+-- @return table? A table with the contents of the GLua global "CustomShipments", or nil if it doesn't exist.
+function darkrp_library.getCustomShipments()
+	return CustomShipments and instance.Sanitize(CustomShipments) or nil
+end
+
+--- Get whether a DarkRPVar is blacklisted from being read by Starfall.
+-- @param string var The name of the variable
+-- @return boolean If the variable is blacklisted
+function darkrp_library.isDarkRPVarBlacklisted(k)
+	checkluatype(k, TYPE_STRING)
+	return not not SF.BlacklistedDarkRPVars[k]
```

Comment:
not not?

Reply:
Oh, so it doesn't return nil. Maybe `SF.BlacklistedDarkRPVars[k]==true` would be easier to understand.